test/quantization/quantize_/workflows/int8/test_int8_tensor.py

```diff
@@ -0,0 +1,243 @@
+
+        self.assertEqual(linear.weight.scale.shape, (N,))
+        self.assertEqual(linear.weight.scale.ndim, 1)
+
+    @common_utils.parametrize(
+        "config",
+        [
+            Int8DynamicActivationInt8WeightConfig(version=2),
+            Int8WeightOnlyConfig(version=2),
+        ],
+    )
+    @common_utils.parametrize("device", ["cpu", "cuda"])
+    @common_utils.parametrize("dtype", [torch.bfloat16, torch.float16])
+    def test_slice(self, config, device, dtype):
+        """Test tensor slicing with per-row quantization"""
+        tensor_size = 256
+        slice_sizes = (64, 128)
+
+        dummy = torch.nn.Linear(
+            tensor_size, tensor_size, bias=False, dtype=dtype, device=device
+        )
+        quantize_(dummy, config)
+
+        weight1 = dummy.weight.clone().narrow(0, 0, slice_sizes[0])
+        weight2 = dummy.weight.clone().narrow(1, 0, slice_sizes[1])
+
+        self.assertEqual(weight1.qdata, dummy.weight.qdata.narrow(0, 0, slice_sizes[0]))
+        self.assertEqual(weight2.qdata, dummy.weight.qdata.narrow(1, 0, slice_sizes[1]))
+        self.assertEqual(weight1.scale, dummy.weight.scale.narrow(0, 0, slice_sizes[0]))
+        self.assertEqual(weight2.scale, dummy.weight.scale)
+        with self.assertRaises(NotImplementedError):
+            _ = dummy.weight[::2]
+
+    @common_utils.parametrize(
+        "config",
+        [
+            Int8DynamicActivationInt8WeightConfig(version=2),
+            Int8WeightOnlyConfig(version=2),
+        ],
+    )
+    def test_index_select(self, config):
+        """test that `x_0 = x[0]` works when `x` is a 2D quantized tensor."""
+        N, K = 256, 512
+        x = torch.randn(N, K, device="cuda", dtype=torch.bfloat16)
+        linear = torch.nn.Linear(K, N, bias=False, dtype=torch.bfloat16, device="cuda")
+        linear.weight.data = x
+        quantize_(linear, config)
+
+        x_int8 = linear.weight
+        x_int8_0 = x_int8[0]
+        torch.testing.assert_close(
+            x_int8.dequantize()[0], x_int8_0.dequantize(), atol=0, rtol=0
+        )
+
+    @common_utils.parametrize(
+        "config",
+        [
+            Int8DynamicActivationInt8WeightConfig(version=2),
+            Int8WeightOnlyConfig(version=2),
+        ],
+    )
+    def test_dequantization_accuracy(self, config):
+        """Test dequantization accuracy separately"""
+        test_data = torch.tensor([[1.0, -1.0]], dtype=torch.bfloat16, device="cuda")
+        linear = torch.nn.Linear(2, 1, bias=False, dtype=torch.bfloat16, device="cuda")
+        linear.weight.data = test_data
+        quantize_(linear, config)
+
+        tensor = linear.weight
+        dequantized = tensor.dequantize()
+        self.assertEqual(dequantized.shape, test_data.shape)
+        assert compute_error(dequantized, test_data) > 20, (
+            f"Dequantization error is too high to get a SQNR of {compute_error(dequantized, test_data)}"
+        )
+
+    @common_utils.parametrize(
```

Comment:
does this have to be parametrize? I think what we need here is to check the code contains a sequence of ops / kernel calls, like this: https://github.com/pytorch/ao/blob/b4ec4cbab799b5320fc1ca094e0398b3ad967dbf/test/quantization/quantize_/workflows/float8/test_float8_tensor.py#L827-L831

I think we can check 1. the quantize op and then 2. the mm op `extern_kernels._int_mm`, in a single run (see example), that should be enough